framework/sandstone.cpp

```diff
@@ -1965,120 +1965,35 @@ static TestResult run_one_test_once(const struct test *test)
 }
 
 static void analyze_test_failures(const struct test *test, int fail_count, int attempt_count,
-                                  const SandstoneApplication::PerThreadFailures &per_thread_failures)
+                                  const PerThreadFailures &per_thread_failures)
 {
     logging_printf(LOG_LEVEL_VERBOSE(1), "# Test failed %d out of %d times"
                                          " (%.1f%%)\n", fail_count, attempt_count,
                    fail_count * 100.0 / attempt_count);
     logging_restricted(LOG_LEVEL_QUIET, "Test failed (#%s%x).", test->id, fail_count - 1);
 
-    // First, determine if all CPUs failed the exact same way
-    bool all_cpus_failed_equally = true;
+    // First, determine if all threads failed the exact same way
+    bool all_threads_failed_equally = true;
     uint64_t fail_pattern = 0;
     int nfailures = 0;
-    for (size_t i = 0; i < num_cpus() && all_cpus_failed_equally; ++i) {
+    for (size_t i = 0; i < num_cpus() && all_threads_failed_equally; ++i) {
         if (per_thread_failures[i]) {
             if (++nfailures == 1)
                 fail_pattern = per_thread_failures[i];
```

Comment:
Oops, bug here. Commit 3db9c3914b80e797fc879bfbe96447dc9bb44207 forgot to change `fail_pattern` type to `__uint128` too. Change it to `PerThreadFailures::value_type`.